### DISCUSSION TITLE
Strange behaviour jest test

### DISCUSSION THREAD

**Discussion by chornos13** [TanStack/query#2758] [Q&A]
Hi All, I want to try contribute in this library.

So, I searched for the issue in the repo and found a bug issue #2741 , I tried to fix the bug but when I made a test, the test always passed which is not expected.

running with
```
yarn test:dev devtools
```

```tsx
// src/devtools/tests/devtools.test.tsx
describe('ReactQueryDevtools', () => {
  ....
  it('should show a disabled label when devtools is open and all observers are disabled', async () => {
    const { queryClient } = createQueryClient()

    function Page() {
      const [enabled, setEnabled] = React.useState(true)
      const { data } = useQuery(
        'key',
        async () => {
          await sleep(10)
          return 'test'
        },
        {
          enabled,
        }
      )

      return (
        <div>
          <h1>{data}</h1>
          <button onClick={() => setEnabled(false)}>Disable Query</button>
        </div>
      )
    }

    renderWithClient(queryClient, <Page />, {
      initialIsOpen: true,
    })

    await waitFor(() =>
      expect(screen.queryByText(/disabled/i)).not.toBeInTheDocument()
    )

    await act(async () => {
      fireEvent.click(await screen.findByText(/disable query/i))
    })

    // commented because I want to make sure the test is fail by using random text and not related to my machine
    // await screen.findByText(/disabled/i)

    console.log('find by text')
    await screen.findByText(/---no matching text here---/i)
    console.log('found text !')
  })
})
```
I expect this test is fail but why the test still passed ?

I try added console.log to make sure and found that second console.log never appear which is because findByText function throw an error

![image](https://user-images.githubusercontent.com/20974979/135748917-133fb664-5330-4546-9909-502166208102.png)

anyone know what's wrong with this ?

btw I'm using Ubuntu 20.04 and all my project using jest is fine except this one

**Comment by TkDodo** [comment]
Maybe we need to use `expect` like above? I’m also curious how you fixed the actual bug because I think it’s not a trivial fix 😅

**Reply by chornos13** [reply]
nope, still passed, and as far as I know `findByText` throw an error like `expect` does so I think I don't need to wrap in `expect`.
![image](https://user-images.githubusercontent.com/20974979/135750630-a6d5ae9c-e192-42b5-a760-aa975623b691.png)

btw I also found a test in devtools that uses `findBy` and it is not wrapped in `expect` in the end line of test 😅 
```tsx

  it('should display the query hash and open the query details', async () => {
    const { queryClient, queryCache } = createQueryClient()

    function Page() {
      const { data = 'default' } = useQuery('check', async () => {
        await sleep(10)
        return 'test'
      })

      return (
        <div>
          <h1>{data}</h1>
        </div>
      )
    }

    renderWithClient(queryClient, <Page />)

    fireEvent.click(
      screen.getByRole('button', { name: /open react query devtools/i })
    )

    const currentQuery = queryCache.find('check')

    await screen.findByText(getByTextContent(`1${currentQuery?.queryHash}`))

    fireEvent.click(
      screen.getByRole('button', {
        name: `Open query details for ${currentQuery?.queryHash}`,
      })
    )

    await screen.findByText(/query details/i)
  })
```

I know this bug is not a trivial as you commented on that issue, I'm just having fun because I'm a fan react-query 😂
